File: fib.java
//I'm changing something, wooooo!
public class fib{
    private static int fib(int x){
	if(x < 2){
	    return x;
	}else{
	    int qianVal = 0;
	    int zuoVal = 1;
	    int res = 0;
	    for(int i = 1; i < x; i++){
		res = qianVal+zuoVal;
	        qianVal = zuoVal;
		zuoVal = res;
	    }
	    return res;
	}
    }
    private static int doWork(int rem){
	int acc = 0;
	for(int i = 0; i < rem; i++){
	    acc += fib((acc+1)%50);
	}
	return acc;
    }
    public static void main(String[] args){
	long start = System.currentTimeMillis();
	System.out.printf("%d\n", doWork(Integer.parseInt(args[0])));
	long duration = System.currentTimeMillis() - start;
	System.out.println("LANGUAGE  Java  "+ duration);
    }
}

File: fib.lua
local function fibRec(qianVal, zuoVal, rem)
      if rem < 1 
      	 then return zuoVal
     	 else return fibRec(zuoVal,qianVal+zuoVal,rem-1)
      end
end

local function fib(x)
      if x < 2 
      	 then return x
	 else return fibRec(0,1,x-1)
      end
end

local function doWork(acc,rem)
      if rem < 1
      	 then return acc
	 else return doWork(acc + fib((acc+1)%50), rem-1)
      end
end

local function start()
      local n = tonumber(arg[1])
      start = os.clock()
      io.stdout:write(doWork(0,n) .. "\n")
      duration = os.clock() - start
      io.stdout:write("LANGUAGE  LuaJit  " .. math.floor(duration*1000) .. "\n")
end

start()